src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -58,24 +92,76 @@ class HttpGoogleCloudStorageDAO(clientSecretsJson: String, dataStoreFactory: Dat
       .execute()
   }
 
-  override def getACL(userId: String, bucketName: String): String = {
-    import system.dispatcher
-    val credential = getCredential(userId)
-    val pipeline = addHeader("Authorization",s"Bearer ${credential.getAccessToken}") ~> sendReceive ~> unmarshal[String]
-    Await.result(pipeline(Get(s"https://www.googleapis.com/storage/v1/b/${bucketName}/acl")),Duration.Inf)
+  override def getMaximumAccessLevel(userId: String, workspaceName: WorkspaceName): GCSAccessLevel = {
+    val credential = getServiceAccountCredential
+    val directory = new Directory.Builder(httpTransport, jsonFactory, credential).setApplicationName(appName).build()
+
+    // note that the iteration order is important here...
+    Seq(GCSAccessLevel.Owner, GCSAccessLevel.Write, GCSAccessLevel.Read) foreach { accessLevel =>
+      try {
+        directory.members().get(makeGroupId(workspaceName, accessLevel), userId).execute()
+        return accessLevel
+      } catch {
+        case e: GoogleJsonResponseException => // fall through to next level
+      }
+    }
+
+    GCSAccessLevel.NoAccess
+  }
+
+  override def getACL(userId: String, bucketName: String, workspaceName: WorkspaceName): Map[String, GCSAccessLevel] = {
+    val credential = getUserCredential(userId)
+    val directory = new Directory.Builder(httpTransport, jsonFactory, credential).setApplicationName(appName).build()
+
+    // TODO should we check our assumption that these groups have the roles we expect them to?
```

Comment:
Yeah. I think we probably don't need to check it here, as long as there's a test that creates the three groups and verifies their ACLs